Lib/sqlite3/test/userfunctions.py

```diff
@@ -351,6 +294,42 @@ def test_empty_blob(self):
         cur = self.con.execute("select isblob(x'')")
         self.assertTrue(cur.fetchone()[0])
 
+    def test_nan_float(self):
+        cur = self.con.execute("select boomerang(?)", (float('nan'),))
```

Comment:
In this test we should check that parameter is substituted with None. isnone would be good.

Add also test for function returning nan (similar to test_func_return_too_large_int).